src/core/public/http/anonymous_paths.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { HttpSetup } from 'src/core/public';
+
+export class AnonymousPaths {
+  private paths: Set<string>;
+
+  constructor(private basePath: HttpSetup['basePath']) {
+    this.paths = new Set();
+  }
+
+  public isAnonymous(path: string): boolean {
+    const pathWithoutBasePath = this.basePath.remove(path);
+    return this.paths.has(this.normalizePath(pathWithoutBasePath));
+  }
+
+  public register(path: string) {
+    if (!path.startsWith('/')) {
```

Comment:
Could `normalizePath` handle this?

Reply:
It definitely can. It felt somewhat awkward to allow consumers to do `anonymousPaths.register('login')` and felt more explicit to require them to do `anonymousPaths.register('/login')`. Requiring the path to start with `/` makes it more obvious that the paths are from the "root" as opposed to relative to anything. This is largely subjective and since it's a platform API, my opinion is just an opinion and I will make it however you prefer.

Reply:
I'm all for making things more robust if it's easy to change. Don't consider it a blocker for this PR though